### PR TITLE
megacmd: Get rid of references to `MEGAjvm`

### LINKED ISSUE #8708: megacmd "CANNOT LINK EXECUTABLE" error
CANNOT LINK EXECUTABLE "mega-cmd-server": cannot locate symbol "MEGAjvm" referenced by  "/data/data/com.termux/files/usr/lib/libmega.so"...
Unable to connect to service: error=2
Please ensure mega-cmd-server is running

termux-info:
Application version:
0.118.0
Packages CPU architecture:
aarch64
...
Android version: 12
Kernel build information:
Linux localhost 5.10.90-v8-g8eb17f0ef62c #1 SMP PREEMPT Fri Jan 14 11:31:47 EET 2022 aarch64 Android
Device manufacturer:
Raspberry
Device model:
Raspberry Pi 4

Same error with  a generic Android 10 tablet.

### PR DESCRIPTION
**Pull request by xtkoba**
Fixes #8708.

### CODE REVIEW

File: packages/megacmd/sdk-src-posix-fs.cpp.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--- a/sdk/src/posix/fs.cpp
++++ b/sdk/src/posix/fs.cpp
+@@ -30,7 +30,7 @@
+ #include "mega/osx/osxutils.h"
+ #endif
+ 
+-#ifdef __ANDROID__
++#ifdef __ANDROID_NO_TERMUX__
```
Comment (thunder-coding) [inline packages/megacmd/sdk-src-posix-fs.cpp.patch:8]:
```suggestion
+#if defined(__ANDROID__) && !defined(__TERMUX__)
```

File: packages/megacmd/sdk-src-posix-fs.cpp.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ #include <jni.h>
+ extern JavaVM *MEGAjvm;
+ #endif
+@@ -1530,7 +1530,7 @@
+ 
+ void PosixFileSystemAccess::statsid(string *id) const
+ {
+-#ifdef __ANDROID__
++#ifdef __ANDROID_NO_TERMUX__
```
Comment (thunder-coding) [inline packages/megacmd/sdk-src-posix-fs.cpp.patch:17]:
```suggestion
+#if defined(__ANDROID__) && !defined(__TERMUX__)
```

File: packages/megacmd/sdk-src-posix-net.cpp.patch [hunk 3]
```diff
@@ -0,0 +1,29 @@
+--- a/sdk/src/posix/net.cpp
++++ b/sdk/src/posix/net.cpp
+@@ -23,7 +23,7 @@
+ #include "mega/posix/meganet.h"
+ #include "mega/logging.h"
+ 
+-#if defined(__ANDROID__) && ARES_VERSION >= 0x010F00
++#if defined(__ANDROID_NO_TERMUX__) && ARES_VERSION >= 0x010F00
```
Comment (thunder-coding) [inline packages/megacmd/sdk-src-posix-net.cpp.patch:8]:
```suggestion
+#if defined(__ANDROID__) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
```

File: packages/megacmd/sdk-src-posix-net.cpp.patch [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ #include <jni.h>
+ extern JavaVM *MEGAjvm;
+ #endif
+@@ -288,7 +288,7 @@
+         ares_library_init(ARES_LIB_INIT_ALL);
+     }
+ 
+-#if defined(__ANDROID__) && ARES_VERSION >= 0x010F00
++#if defined(__ANDROID_NO_TERMUX__) && ARES_VERSION >= 0x010F00
```
Comment (thunder-coding) [inline packages/megacmd/sdk-src-posix-net.cpp.patch:17]:
```suggestion
+#if defined(__ANDROID__) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
```

File: packages/megacmd/sdk-src-posix-net.cpp.patch [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+     initialize_android();
+ #endif
+ 
+@@ -2852,7 +2852,7 @@
+     return (DNS_CACHE_EXPIRES && (Waiter::ds - ipv6timestamp) >= DNS_CACHE_TIMEOUT_DS);
+ }
+ 
+-#if defined(__ANDROID__) && ARES_VERSION >= 0x010F00
++#if defined(__ANDROID_NO_TERMUX__) && ARES_VERSION >= 0x010F00
```
Comment (thunder-coding) [inline packages/megacmd/sdk-src-posix-net.cpp.patch:26]:
```suggestion
+#if defined(__ANDROID__) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
```

**Review by thunder-coding — CHANGES_REQUESTED** [review]
You always seem to have fix for every thing. :)

Just a few changes to maintain consistency

File: packages/megacmd/sdk-src-posix-net.cpp.patch (hunk 5 above)
Comment (xtkoba) [reply packages/megacmd/sdk-src-posix-net.cpp.patch]: Thanks!

**Review by thunder-coding — APPROVED** [review]

**Pull request merged**